AdamantTests/Modules/Wallets/DashWalletServiceTests.swift

```diff
@@ -0,0 +1,363 @@
+    
+    private var sut: DashWalletService!
+    private var lastTransactionStorageMock: DashLastTransactionStorageProtocolMock!
+    private var apiServiceMock: DashApiServiceProtocolMock!
+    private var apiCoreMock: APICoreProtocolMock!
+    private var transactionFactoryMock: BitcoinKitTransactionFactoryProtocolMock!
+    
+    override func setUp() {
+        super.setUp()
+        
+        lastTransactionStorageMock = DashLastTransactionStorageProtocolMock()
+        transactionFactoryMock = BitcoinKitTransactionFactoryProtocolMock()
+        transactionFactoryMock.stubbedTransactionFactory = {
+            BitcoinKit.Transaction.createNewTransaction(
+                toAddress: $0,
+                amount: $1,
+                fee: $2,
+                changeAddress: $3,
+                utxos: $4,
+                lockTime: $5,
+                keys: $6
+            )
+        }
+        apiCoreMock = APICoreProtocolMock()
+        apiServiceMock = DashApiServiceProtocolMock()
+        apiServiceMock.api = DashApiCore(apiCore: apiCoreMock)
+        sut = DashWalletService()
+        
+        sut.dashApiService = apiServiceMock
+        sut.lastTransactionStorage = lastTransactionStorageMock
+        sut.addressConverter = makeAddressConverter()
+        sut.transactionFactory = transactionFactoryMock
+    }
+    
+    override func tearDown() {
+        sut = nil
+        apiServiceMock = nil
+        lastTransactionStorageMock = nil
+        apiCoreMock = nil
+        transactionFactoryMock = nil
+        
+        super.tearDown()
+    }
+    
+    func test_createTransaction_throwsErrorWhenHasLastTransactionIdAndNotEnoughConfirmations() async throws {
+        // given
+        lastTransactionStorageMock.stubbedLastTransactionId = Constants.lastTransactionId
+        await apiCoreMock.isolated { mock in
+            mock.stubbedSendRequestBasicResult = APIResponseModel(
+                result: .success(Constants.getTransactionZeroConfirmationsData),
+                data: Constants.getTransactionZeroConfirmationsData,
+                code: 200
+            )
+        }
+        
+        // when
+        let result = await Result {
+            try await self.sut.create(recipient: Constants.invalidDashAddress, amount: 10)
+        }
+        
+        // then
+        switch result.error as? WalletServiceError {
+        case .remoteServiceError?:
+            break
+        default:
+            XCTFail("Expected .remoteServiceError, but got \(result.error) error")
+        }
+    }
+    
+    func test_createTransaction_throwsErrorWhenNoWallet() async throws {
+        // given
+        lastTransactionStorageMock.stubbedLastTransactionId = nil
+        
+        // when
+        let result = await Result {
+            try await self.sut.create(recipient: Constants.invalidDashAddress, amount: 10)
+        }
+        
+        // then
+        XCTAssertEqual(result.error as? WalletServiceError, .notLogged)
```

Comment:
What do you think of using `XCTAssertThrowsError` ?

Reply:
I prefer to use consistent logic for handling fail-case scenarios and success scenarios, so code looks the same

```
// fail
XCTAssertEqual(result.error as? WalletServiceError, .notLogged)
// success
XCTAssertEqual(result.value, expectedResult)

```